### DISCUSSION TITLE
KEM list mismatch

### DISCUSSION THREAD

**Discussion by sumantasarkar** [open-quantum-safe/liboqs#934] [Q&A]
I was calling KEMs one by one from the list of schemes by  OQS_KEM_alg_identifier(scheme_id). It was supposed to call SIKE-p751 as the 63rd scheme, however, it called SIKE-p610-compressed instead. I found that SIKE-p751 was actually referring to the 60th scheme in that list. Am I working with an older list, or missing something?

Thanks
Sumanta

**Comment by baentsch** [comment]
Not sure I understand exactly what you're doing: Can you share some code to reproduce the issue? Also, what leads you to state 

> It was supposed to call SIKE-p751 as the 63rd scheme

? Did you check/validate alg enablement via `OQS_KEM_alg_is_enabled`? Did you build liboqs yourself or are you using a pre-build docker image? The latter would help us assure we have the same configuration when reproducing the issue.

**Comment by sumantasarkar** [comment]
I downloaded liboqs-0.4.0 which is the latest release. OQS_KEM_alg_is_enabled is done. 
> Also, what leads you to state
Actually I was testing KEMs one by one, I found that there is a list of algorithms in oqs/kem.h. Once I write OQS_KEM_alg_identifier(size_t i), it calls the i-th scheme, so varying i could test all the KEMs listed there. But I got a mismatch in calling the 63rd scheme. I agree that this is not serious, just rearranging the list will solve the issue. I just wanted to draw the attention of the contributors about this.

**Reply by baentsch** [reply]
I still can't reproduce the issue. Please share some code.
